java/ql/test/library-tests/frameworks/stapler/DataBoundPostConstructTest.java

```diff
@@ -0,0 +1,42 @@
+import javax.annotation.PostConstruct;
+import net.sf.json.JSONObject;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundResolvable;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+
+public class DataBoundPostConstructTest implements DataBoundResolvable {
+
+    static Object source(String label) {
+        return null;
+    }
+
+    static void sink(Object o) {}
+
+    static void test() {
+        new DataBoundPostConstructTest(source("constructor"));
+        new DataBoundPostConstructTest(null).setField(source("setter"));
```

Comment:
Here we are relying on `source` rather than the MaD-declared remote sources, which was a bit confusing to me.

Reply:
These were supposed to test the dataflow steps rather than the sources (the MaD models don't have dedicated tests). And using the `source` method allowed me to label the sources so that I could verify that flow from both reaches the sinks.

But I know this could be done using the MaD sources by altering the QL test, so if you think that'd be less confusing I'll do that.

Reply:
It's fine as-is.